src/cwe_checker_lib/src/analysis/taint/state/tests.rs

```diff
@@ -58,6 +60,69 @@ fn new_pointer(location: &str, offset: i64) -> DataDomain<ValueDomain> {
     DataDomain::from_target(id, bv(offset))
 }
 
+// FIXME: This illustrates the current, unsound merging of memory taints. Make
+// sure to change this test when you work on a better memory model.
+#[test]
+fn merge_memory_object_overlapping() {
+    let taint_8 = Taint::Tainted(ByteSize::new(8));
+    let taint_4 = Taint::Tainted(ByteSize::new(4));
+
+    let mut memory_object = MemRegion::<Taint>::new(ByteSize::new(8));
+    let mut other_memory_object = MemRegion::<Taint>::new(ByteSize::new(8));
+
+    memory_object.insert_at_byte_index(taint_4, 2);
+    memory_object.insert_at_byte_index(taint_8, 8);
+    memory_object.insert_at_byte_index(taint_4, 16);
+    other_memory_object.insert_at_byte_index(taint_8, 0);
+    other_memory_object.insert_at_byte_index(taint_4, 8);
+    other_memory_object.insert_at_byte_index(taint_4, 14);
+
+    merge_memory_object_with_offset(&mut memory_object, &other_memory_object, 0);
```

Comment:
From my experience, it is important to also test the case when there is an actual offset other than zero. Just to be sure that offset computations are done without sign errors.